apps/etl-func/src/adapters/__tests__/message-event.test.ts

```diff
@@ -0,0 +1,76 @@
+import { aSimpleMessageEvent } from "@/__mocks__/message-event.js";
+import { Logger } from "pino";
+import { beforeEach, describe, expect, test, vi } from "vitest";
+
+import { messageSchema } from "../avro.js";
+import { EventHubEventProducer } from "../message-event.js";
+
+const mocks = vi.hoisted(() => ({
+  EventHubProducerClient: vi.fn().mockImplementation(() => ({
+    createBatch: () => ({
+      tryAdd: tryAddMock,
+    }),
+    sendBatch: sendBatchMock,
+  })),
+}));
+
+vi.mock("@azure/event-hubs", async (importOriginal) => {
+  const original = await importOriginal<typeof import("@azure/storage-blob")>();
+  return {
+    ...original,
+    EventHubProducerClient: mocks.EventHubProducerClient,
+  };
+});
+
+const tryAddMock = vi.fn(() => true);
+const sendBatchMock = vi.fn(() => Promise.resolve());
+
+const eventHubProducerClient = new mocks.EventHubProducerClient();
+
+const errorLogMock = vi.fn();
+
+const loggerMock = {
+  error: errorLogMock,
+} as unknown as Logger;
+
+const messageEventAdapter = new EventHubEventProducer(
+  eventHubProducerClient,
+  messageSchema,
+  loggerMock,
+);
+
+describe("publishMessageEvent", () => {
+  beforeEach(() => {
+    tryAddMock.mockClear();
+    tryAddMock.mockImplementation(() => true);
+    sendBatchMock.mockClear();
+    sendBatchMock.mockImplementation(() => Promise.resolve());
+  });
```

Comment:
```suggestion
  beforeEach(() => {
    tryAddMock.mockRestore();
    sendBatchMock.mockRestore();
  });
```

`mockRestore` handles the `clear` and restore the implementation of the mock to it's original